solr/bin/solr

```diff
@@ -2071,6 +2071,17 @@ else
   REMOTE_JMX_OPTS=()
 fi
 
+# Enable java security manager (limiting filesystem access and other things)
+# Don't cache DNS lookups forever, set the value back to the original JDK default.
+if [ "$SOLR_SECURITY_MANAGER_ENABLED" == "true" ]; then
+  SECURITY_MANAGER_OPTS=('-Djava.security.manager' \
+    "-Djava.security.policy=${SOLR_SERVER_DIR}/etc/security.policy" \
+    '-Dsun.net.inetaddr.ttl=30' \
```

Comment:
Shouldn't this be configurable by the user? Perhaps we keep this default but make it configurable with a `SOLR_NETWORKADDRESS_CACHE_TTL` variable.

Although most (all?) JVMs still honor the `sun.net.inetaddr.ttl` but it is officially deprecated so we should use `-Dnetworkaddress.cache.ttl`

Reply:
{quote}
Shouldn't this be configurable by the user? Perhaps we keep this default but make it configurable with a SOLR_NETWORKADDRESS_CACHE_TTL variable.
{quote}

It is not configurable today: I'm not trying to add that feature here. I am just preventing dns from cached indefinitely.

{quote}
Although most (all?) JVMs still honor the sun.net.inetaddr.ttl but it is officially deprecated so we should use -Dnetworkaddress.cache.ttl
{quote}

nope: `networkaddress.cache.ttl` is not a system property at all.

Reply:
@shalinmangar i added a commit with a custom `java.security.properties` file, to override this via the `networkaddress.cache.ttl`, setting it back to the default of 30 (same as when no security manager is active). It does the same thing as before, just in a more roundabout way...

If users want to mess with java security properties, they must edit java security properties files either way: no variables can be supported since they are not system properties and can't be set with `-D`.